bootstrap/config/manager/base/deployment.yaml

```diff
@@ -73,6 +60,13 @@ spec:
         image: controller:latest
         imagePullPolicy: Always
         name: manager
+        env:
```

Comment:
Wouldn't it be easier if we have the base layer without these envs, and just add them in the overlays? This way if we add other env vars the order would not matter.
Also, we should probably disable this behaviour by default and assume public addresses. If that's not the case the user should know and configure the provider accordingly (I know the main use case is with this feature enabled, but I feel we should enable it when we declare what address we have, as by default we would need to fail the controller if it's enabled but not internal namespace/service it's defined)

Reply:
Regardless what default value to choose, the usage of the env var needs to be documented in the  README.md.

Reply:
I did initially try that (no env vars in the base and add them in overlay) but kustomize acted weird with it :/ I might've been patching it incorrectly though